ui/src/main/webapp/src/app/reports/technologies/technologies-report-ejb.component.html

```diff
@@ -83,113 +74,75 @@ <h3 class="panel-title">Message Driven Beans</h3>
-                    <thead wu-sortable-table
-                           [(sortedData)]="sortedEjbEntityBean" [data]="filteredEjbEntityBean"
-                           [tableHeaders]="[
-                            { title: 'Bean Name', isSortable: true, sortBy: 'name' },
-                            { title: 'Interface', isSortable: true, sortBy: 'interface'},
-                            { title: 'Implementation', isSortable: true, sortBy: 'class'},
-                            { title: 'JNDI Location', isSortable: true, sortBy: 'location'}
-                        ]">
-                    </thead>
-                    <tbody>
-                    <tr *ngFor="let value of this.sortedEjbEntityBean">
-                        <td class="col-md-2">{{value.name}}</td>
-                        <td class="col-md-2">{{value.interface}}</td>
-                        <td class="col-md-6"><a [routerLink]="['../source/' + value.sourceVertexId]">{{value.class}}</a></td>
-                        <td class="col-md-2">{{value.location}}</td>
-                    </tr>
-                    </tbody>
-                </table>
+    <wu-table
+            [title]="'Stateful Session Beans'"
+            [items]="ejbSessionStatefulBean"
+            (clearFilter)="clearSearch()"
+            [filter]="filterCallback"
+            [loading]="loadingEJBStateful"
+            [tableHeaders]="[
+                { title: 'Bean Name', isSortable: true, sortBy: 'name' },
+                { title: 'Interface', isSortable: true, sortBy: 'interface'},
+                { title: 'Implementation', isSortable: true, sortBy: 'class'},
+                { title: 'JNDI Location', isSortable: true, sortBy: 'location'}
+            ]"
+            [rowTemplate]="rowTemplateEjbStateless"
+    >
+        <ng-template #rowTemplateEjbStateless let-value>
+            <tr>
+                <td class="col-md-2">{{value.name}}</td>
+                <td class="col-md-2">{{value.interface}}</td>
+                <td class="col-md-6"><a [routerLink]="['../source/' + value.sourceVertexId]">{{value.class}}</a></td>
+                <td class="col-md-2">{{value.location}}</td>
+            </tr>
+        </ng-template>
+    </wu-table>
 
-            </wu-all-data-filtered-message>
-        </div>
-    </div>
+    <wu-table
+            [title]="'Entity Beans'"
+            [items]="ejbEntityBean"
+            (clearFilter)="clearSearch()"
+            [filter]="filterCallback"
+            [loading]="loadingEntity"
+            [tableHeaders]="[
+                { title: 'Bean Name', isSortable: true, sortBy: 'name' },
+                { title: 'Interface', isSortable: true, sortBy: 'interface'},
+                { title: 'Implementation', isSortable: true, sortBy: 'class'},
+                { title: 'JNDI Location', isSortable: true, sortBy: 'location'}
+            ]"
+            [rowTemplate]="rowTemplateEjbStateless"
+    >
+        <ng-template #rowTemplateEjbStateless let-value>
```

Comment:
This template is repeated 3 times so maybe it could be placed at the end of the component, where the previous `#loadingData` template was placed.